app/main.py

```diff
@@ -1,6 +1,20 @@
 from typing import Callable
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> any:
+        key = (args, tuple(sorted(kwargs.items())))
+        if key in cached_results:
+            print("Getting from cache")
+            return cached_results[key]
+        else:
```

Comment:
There is no need for `else` since the code will not execute after the `return` statement anyway. Please, remove the redundant `else`